presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationUtils.java

```diff
@@ -268,15 +268,24 @@ public static void mergeRegressionState(RegressionState state, RegressionState o
     public static String generateAggregationName(String baseName, TypeSignature outputType, List<TypeSignature> inputTypes)
     {
         StringBuilder sb = new StringBuilder();
-        sb.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, outputType.toString()));
+        sb.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, getAbbreviatedTypeName(outputType)));
         for (TypeSignature inputType : inputTypes) {
-            sb.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, inputType.toString()));
+            sb.append(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_CAMEL, getAbbreviatedTypeName(inputType)));
```

Comment:
This can potentially generate the same name for different functions. Are we ok when collision happens? e.g.

AccumulatorCompiler.java:L155
```
makeClassName(metadata.getName() + accumulatorInterface.getSimpleName()),
```
We could potentially define classes with the same name, but it won't be a problem for JVM?

Reply:
`makeClassName` internally actually use an auto-inc ID so that will guarantee uniqueness. Bytecode gen should not depend on the pass in class name being unique to generate correct code anyways.

Reply:
For reference: https://github.com/prestodb/presto/pull/15711